staging/vhost-device-gpu/src/vhu_gpu.rs

```diff
@@ -394,28 +374,29 @@ impl VhostUserGpuBackend {
         writer: &mut Writer,
         signal_used_queue: &mut bool,
     ) -> Result<()> {
-        let mut response = Err(ErrUnspec);
-
-        let (ctrl_hdr, gpu_cmd) = match GpuCommand::decode(reader) {
-            Ok((ctrl_hdr, ctrl_cmd)) => {
-                response = self.process_gpu_command(virtio_gpu, ctrl_hdr, ctrl_cmd);
-                (Some(ctrl_hdr), Some(ctrl_cmd))
+        let mut response = ErrUnspec;
+
+        let ctrl_hdr = match GpuCommand::decode(reader) {
+            Ok((ctrl_hdr, gpu_cmd)) => {
+                // TODO: consider having a method that return &'static str for logging purpose
+                let cmd_name = format!("{:?}", gpu_cmd);
+                let response_result = self.process_gpu_command(virtio_gpu, ctrl_hdr, gpu_cmd);
+                // Unwrap the response from inside Result and log information
+                response = match response_result {
+                    Ok(response) => {
+                        trace!("GpuCommand {cmd_name} success: {response:?}");
+                        response
+                    }
+                    Err(response) => {
+                        debug!("GpuCommand {cmd_name} failed: {response:?}");
+                        response
+                    }
+                };
+                Some(ctrl_hdr)
             }
             Err(e) => {
                 warn!("Failed to decode GpuCommand: {e}");
-                (None, None)
-            }
-        };
-
-        // Unwrap the response from inside Result and log information
-        let mut response = match response {
-            Ok(response) => {
-                trace!("GpuCommand {gpu_cmd:?} success: {response:?}");
-                response
-            }
-            Err(response) => {
-                debug!("GpuCommand {gpu_cmd:?} failed: {response:?}");
-                response
+                None
```

Comment:
Instead of returning some and none, I suggest to handle the ase when decoding fails, atleast return some error of some sort, maybe `ErrUnspec`

Reply:
This was my first instinct too, and I have originally written it this way, but I think it makes sense to execute other commands even if the guest specified an invalid/unknown command for us. This is what crossvm does and the original code you had here did too.
If we bubble up the errors, then we would crash I guess.
It depends how forgiving we want to be to the guest. 

The case is handled 
- logging at the debug level (though maybe this should be at least `warn!` level)
- the `ErrUnspec` error is written to the guest, because the command failed

Reply:
yeah, I agree. we should be allowed to execute other commands regardless of one failure, Atleast the guest can recover from failure of one gpu command.